src/plugins/telemetry/public/plugin.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Plugin, CoreSetup, CoreStart, HttpSetup } from '../../../core/public';
+
+import { TelemetrySender, TelemetryService, TelemetryNotifications } from './services';
+
+export interface TelemetryPluginStart {
+  telemetryService: TelemetryService;
+  telemetryNotifications: TelemetryNotifications;
+}
+
+export class TelemetryPlugin implements Plugin<void, TelemetryPluginStart> {
+  private isUnauthenticated: boolean = true;
+
+  public setup({ http }: CoreSetup) {
+    this.isUnauthenticated = this.getIsUnauthenticated(http);
```

Comment:
Is there a way to detect page change through core? I was depending here on the complete page refresh between anonymous and non-anonymous pages.

Reply:
We do not. There is an issue open that has not been prioritized (https://github.com/elastic/kibana/issues/44868). We could expose this pretty easily though.

The only other option I can think of is using the [`onpopstate`](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onpopstate) event. I don't love this option though since it makes testing a bit harder. Let me know what you think and we can prio #44868 if needed.

Reply:
@joshdover I've exposed it in the PR. it does change per app id not per page change, which should be enough for my usecase i believe. App change is always triggered between login/logout/status.

The method `getIsUnauthenticated` is working properly and i've been using it in the `usageCollection` plugin as well.

Reply:
Yeah it's not a perfect mapping right now. It would be better if the anonymous paths logic just exposed an observable of whether or not the user was on an anonymous path currently.

Reply:
I'm happy to update my code in the future if NP changes the api of it

Reply:
#44868 should be trivial, the `currentAppId$` is already exposed in core's internals.